config.py

```diff
@@ -3,7 +3,16 @@
 host = "localhost"
 database = "mapbot"
 port = "3306"
-#key = "*Google_Cloud_API_key*" #Will be provided by mentors
-#stanford_path_to_jar = "*your_path_to_stanford-corenlp-x.x.x.jar*" #for eg. "stanford-corenlp-full-2018-10-05/stanford-corenlp-3.9.2.jar"
-#stanford_path_to_models_jar = "*your_path_to_stanford-corenlp-x.x.x-models.jar*" #for eg. "stanford-corenlp-full-2018-10-05/stanford-corenlp-3.9.2-models.jar"
-#javahome = '*your_path_to_jdk_bin_java.exe*' #for eg. "C:/Program\ Files/Java/jdk1.8.0_201/bin/java.exe"
+key = "*Google_Cloud_API_key*" #Will be provided by mentors
+stanford_path_to_jar = "./stanford-corenlp-full-2018-10-05/stanford-corenlp-3.9.2.jar" #"*your_path_to_stanford-corenlp-x.x.x.jar*"
+stanford_path_to_models_jar = "./stanford-corenlp-full-2018-10-05/stanford-corenlp-3.9.2-models.jar" #"*your_path_to_stanford-corenlp-x.x.x-models.jar*"
+javahome = '*your_path_to_jdk_bin_java.exe*'#for eg. 'C:\\Program\ Files\\Java\\jdk1.8.0_201\\bin\\java.exe' or '/usr/local/openjdk-11/bin/java'
+
+# DONOT CHANGE THE VALUES BELOW DURING INITIAL CONFIGURATION SET UP
+import os
+if os.getenv("DOCKER"):
+    # print("Inside Docker")
+    user = "root"
+    password = "root"
+    host = "db"
+    javahome = '/usr/local/openjdk-11/bin/java'
```

Comment:
If I'm not mistaken, I think these should be specified as environments and environment variables within the Dockerfile itself rather than applying these configs from within our app, if `DOCKER` env is present.

That is, there's a way in which we can specify environment names and variables like ...

```yaml
Environment athityakumar:
  user: athityakumar
  password: password1
  host: db
  javahome: /usr/local/openjdk-8/bin/java

Environment vishakha:
  user: vishakha
  password: password2
  host: db2
  javahome: /usr/local/openjdk-9/bin/java

Environment ankur:
  user: ankur
  password: password3
  host: db3
  javahome: /usr/local/openjdk-11/bin/java
```

... and apply a specific environment (say, ankur) while deploying from the Dockerfile. This is usually done from the context of deploying to different environments like dev/qa/e2e/prd - but you get the point.

This way, our app only does `os.getenv("user")` and Docker takes care of which environment's `user` config has to applied.

Reply:
I've been working on [what](https://github.com/vishakha-lall/MapBot/pull/41#issuecomment-593721745) @vishakha-lall suggested. Would be happy to introduce a different way to have configuration variables. Views @vishakha-lall ?

Reply:
This could be a solution to what we're looking for. Why don't you try this out @chttrjeankr? Although, we might not need multiple environments per se since we do not plan to get rid of `config.py` just yet. But yes, the configs that we are looking for can be incorporated here.

Reply:
> If I'm not mistaken, I think these should be specified as environments and environment variables within the Dockerfile itself rather than applying these configs from within our app, if `DOCKER` env is present.

Do you really mean the `Dockerfile`? I like how clean it looks right now.

Reply:
@athityakumar let me know where do you think environment variables should be added?
- `Dockerfile`
- `docker-compose.yml`
- a separate `.env` file

Reply:
Let us first find the cause to the error we are facing, and then finalise the approach z shall we?

Reply:
Sounds better! :rocket:  I ran bash inside the `mapbot_bot` container and checked for the environment variables and used Python to `getenv()` it. It worked.
![image](https://user-images.githubusercontent.com/39518771/75807758-f508b200-5dab-11ea-977f-dea55d9fa7e4.png)

I don't' get why it's not working for `config.py`

Reply:
![Screenshot from 2020-03-04 00-16-04](https://user-images.githubusercontent.com/39518771/75808914-ddcac400-5dad-11ea-968e-fa388ecb74a2.png)
The parts of the screen are `databaseconnect.py`, `config.py` and the output terminal respectively from top to bottom.

I also tried introducing `print` statements in the programs to check the program flow. Seems to work fine. I used
```
sudo docker-compose up --build -d && sudo docker attach mapbot_python
```
to rebuild and start up the containers.

Reply:
I don't see why we need to rebuild the image, since there are no setup (import configs) that are executed there. I uncommented the "Inside Docker" print statement, but I don't see that, looks like we're missing something very important. 
![image](https://user-images.githubusercontent.com/31643223/75838609-636d6480-5dec-11ea-8b14-4abe106178e0.png)
Can you add print statements to the `config.py` and print values of host in both the flows (with and without Docker), I feel the value is being overwritten. Also, please print the vaue of host in `databaseconnect.py` to ascertain that we are using `db` and not `localhost`.

Reply:
Okay, rebuilding the project works and I figured out why. Since the volume for the bot container is not mounted to the code volume, any new changes do not reflect in the container without rebuilding the image which copies the code directory to the container directory. 
We will create a new issue to address this, as this would make development with containerised application difficult.
Long story short, it looks like all is fine here as far as the scope of this issue is concerned. @janakrajchadha @athityakumar let us push any other reviews to new issues, we already got a lot done here! What do you guys think?

Reply:
Absolutely! Lots of good work in here by @chttrjeankr ! New issues for the pending work from this one and the idea of spinning up the complete app using just the Dockerfile makes more sense.

Reply:
I had that issue in mind that provides a hindrance while developing with Docker but figured it'd be handled in another new issue. :no_mouth: